packages/schematics/angular/app-shell/index.ts

```diff
@@ -237,7 +237,11 @@ function addServerRoutes(options: AppShellOptions): Rule {
     if (!clientServerOptions) {
       throw new SchematicsException('Server target does not contain options.');
     }
-    const modulePath = getServerModulePath(host, clientProject.sourceRoot || 'src', options.main as string);
+    const sourceRoot = clientProject.sourceRoot || 'src';
+    const serverPath = options.appDir ?
+     `${sourceRoot}/${options.appDir}` :
+    sourceRoot;
```

Comment:
This change needs a more thorough analysis. I think that `appDir` option should be removed - it didn't work in the first place, so just deprecate it now and remove it later, and there is no harm done.

In other schematics, [buildDefaultPath()](https://github.com/angular/angular-cli/blob/v10.0.0-rc.0/packages/schematics/angular/utility/workspace.ts#L72-L81) is used to hardcode an `app` or `lib` directory depending on the project type and it will come into conflict with this; you want other schematics to work even for projects that have used the `app-shell` schematics.

Alternatively, this could be implemented for all schematics, and have it fallback to the current behavior of `buildDefaultPath()`. I've actually submitted a similar feature suggestion, but for removing the sub-directory for all schematics, see #17598.

But I'm not sure if either of these should be implemented, we need to make sure that existing tutorials on how to generate components will continue to work in the future (including for projects that have used `app-shell`).